Source/Core/Core/Scripting/ScriptingPlugins_Documentation.md

```diff
@@ -0,0 +1,226 @@
+# Table of Contents
+
+- [About this File](#about)
+- [Folder Structure for Scripting Plugins](#folder-structure)
+- [Functions that Scripting Plugins Need to Export as DLL Functions](#required-functions)
+- [Functions that Scripting Plugins Need to Implement](#implemented-functions)
+- [Creating the DLL](#creation)
+
+---
+
+<div id='about'/>
+
+# About this File:
+
+This file specifies how to create a Scripting Plugin (what files are required to create a Scripting Plugin, what functions need to be created/defined, etc.). To see an example of a completed ScriptingPlugin for Lua, you can check out this link: https://github.com/Lobsterzelda/lobster-zelda-dolphin/tree/lua-support/PluginSources/Lua
+
+If you want to read a description of what each file does which is located in the Source/Core/Core/Scripting folder (which is shared by all DLLs), then you can check out the file in this folder called Common_Scripting_Functions_Documentation.md
+
+---
+
+<div id='folder-structure'/>
+
+# Folder Structure for Scripting Plugins:
+
+All Scripting plugins will get put into the PluginSources folder, which will be put in the root directory of Dolphin (the same folder that contains the Binary folder).
+
+Each language will be contained in a separate folder inside of PluginSources - to reduce ambiguity, this should be the name of the language that's being implemented.
+
+The language folder will contain a Source folder, which contains all of the source code for the plugin. The language folder will also contain a file called extensions.txt, which lists the extension types for that language. If we want all scripts ending in .lua to be treated as Lua scripts, then extensions.txt should just contain the following line:
+
+.lua
+
+As an example, suppose we want to implement the Lua language. Then, our file structure will look like this, with the root directory representing the root directory for Dolphin (D represents directories, F represents files):
+
+/ (D)
+
+----PluginSources (D)
+
+--------Lua (D)
+
+------------extensions.txt (F)
+
+------------Source (D)
+
+----------------CoreScriptInterface (D)
+
+----------------<The source code for the DLL> (Multiple files and or directories).
```

Comment:
This seems difficult to read, how about making this a code block instead?
```suggestion
[/]
+- [PluginSources]
   +- [Lua]
      +- extensions.txt
      +- [Source]
         +- [CoreScriptInterface]
         +- <The source code for the DLL> (Multiple files and or directories).
```

Not really feeling that one either, but maybe we can come up with something more readable together.